src/common/map/MapDirective.js

```diff
@@ -13,6 +13,47 @@
             zoom: '='
           },
           link: function(scope, element) {
+
+            function createVector(coor) {
```

Comment:
You need to move this function into the mapService function. 
We do not know the projection every time. This can be changed. So use the default projection from the map instead of just `EPSG:3857`